urnai/base/savable.py

```diff
@@ -0,0 +1,225 @@
+        file name that should be used while
+        persisting the object.
+        """
+        return self.__class__.__name__ + '_'
+
+    def save_pickle(self, persist_path):
+        """
+        This method saves our instance
+        using pickle.
+
+        First it checks which attributes should be
+        saved using pickle, the ones which are not
+        are backuped.
+
+        Then all unpickleable attributes are set to None
+        and the object is pickled.
+
+        Finally the nulled attributes are
+        restored.
+        """
+        path = self.get_full_persistance_pickle_path(persist_path)
+		
+		# creating necessary directories for the file
+        os.makedirs(os.path.dirname(path), exist_ok=True)
+		
+        with open(path, 'wb') as pickle_out:
+            pickle.dump(self.get_pickleable_dict(), pickle_out)
+
+    def save_extra(self, persist_path):
+        """
+        This method should be implemented when
+        some extra persistence is to be saved.
+        """
+        pass
+
+    def load_pickle(self, persist_path):
+        """
+        This method loads a list instance
+        saved by pickle.
+        """
+        # Check if pickle file exists
+        pickle_path = self.get_full_persistance_pickle_path(persist_path)
+        exists_pickle = os.path.isfile(pickle_path)
+        
+        # If yes, load it
+        if exists_pickle and os.path.getsize(pickle_path) > 0:
+            with open(pickle_path, 'rb') as pickle_in:
+                pickle_dict = pickle.load(pickle_in)
+                self.restore_pickleable_attributes(pickle_dict)
+                '''
+                rp.report(
+                    '**************************************** \n Loaded pickle for '
+                    + self.get_default_save_stamp()
+                    + ' loaded. \n****************************************',
+                2)
+				'''
+
+    def load_extra(self, persist_path):
+        """
+        This method should be implemented when
+        some extra persistence is to be loaded.
+        """
+        pass
+
+    def get_full_persistance_pickle_path(self, persist_path):
+        """This method returns the default persistance pickle path."""
+        return (persist_path + os.path.sep + self.get_default_save_stamp()
+		+ '.pkl')
+
+    def get_full_persistance_tensorflow_path(self, persist_path):
+        """This method returns the default persistance tensorflow path."""
+        return (persist_path + os.path.sep + self.get_default_save_stamp()
+		+ 'tensorflow_')
+
+    def get_full_persistance_path(self, persist_path):
+        """This method returns the default persistance path."""
+        return persist_path + os.path.sep + self.get_default_save_stamp()
+
+    def get_full_persistance_pytorch_path(self, persist_path):
+        """This method returns the default persistance pytorch path."""
+        return (persist_path + os.path.sep + self.get_default_save_stamp()
+		+ '.pt')
```

Comment:
I would remove all these functions, with the exception of `get_full_persistance_path`.